### ISSUE TITLE
General discussion about redcap limitations/organisation/usage

### ISSUE THREAD

**Issue by fredbarthelemy** [INT-NIT/DigLabTools#16]
FRED says:
Hello,
Just for you to be aware of this limitation I found in RedCap. You can't modify the status "partly completed" to "complete" of a record by importing a CSV. This is logical to avoid messing up with the forms but it also means that in our situation the sessions that were recorded in a row will remain labeled as incomplete. The only way I see to work around the problem would be to supress all the entries in redcap and import the csv. But it's not possible to supress all entries at once and I can't imagine to suppress all 990 entries one by one.

An alternative, although it seems stupid, would be to give these sessions the status "not completed" and modify them within the redcap interface.

Another point. For the next users, please tell them not to use excel to edit the CSV. It messes up with dates (applies its default format which may not be understood during the reupload to redcap) and session numbers (00 are suppressed).

**Comment by fredbarthelemy** [comment]
JULIA says:
thanks for the warnings. Regarding the 'partly completed' and 'complete' status issue: I am not sure what you exactly mean by 'suppressing' entries in redcap? Do you mean to delete all records and reimport the adjusted version via csv? I can check again if this is something that could be addressed for now with admin rights, but I don't have high hopes there and this also wouldn't be a solution applicable in multiple cases.

Did you test openoffice calc for editing instead? Does this one do a better job than Excel in terms of not automatically interpreting values? Or would you recommend a plain text editor?

**Comment by fredbarthelemy** [comment]
SYLVAIN says:
My global thoughts:

- if I'm not mistaken most of these problems come from the attempts at editing records a posteriori... at some point, we had decided that it would not be possible because it would be a mess... we've realized that it is necessary for the users and we cannot go around it, but clearly, our anticipation that it would be a mess clearly was good... (not very constructive point... let's see the next one for a more contructive one )

- about your last point ("please tell them etc."), this tells me that we should have a real user documentation, with precise procedures: what is possible to do? what is not possible? how to do each of these things, step by step? etc.; to be constructive on my previous point, I really think it's important to set limits, and to define what is not possible to do! so that we can train users on what's possible to do...

WDYT?

**Comment by fredbarthelemy** [comment]
JULIA says:
I agree that setting limits / responsabilities would make this much easier. I like the idea of the standard user not having to interact with redcap itself as this would make A) their life much easier, B) limit the number of people who need to know about redcap mechanics to a few people (experts / datamanager / fredish people).

Regarding the need for change of records later on there's two cases:

- The current implementation has a feature (more likely a bug) that does not mark all records as complete that are complete. I hope this will be fixed by using a different mechanic after the RedCap server update.

- A mistake during the recording. This should be rare and I think can be handled by approaching one redcap expert / datamanager to fix this single entry.


In this scenario the users would not need to need to use excel / libre office calc for loading the data, assuming they adopt the BIDS approach and do not need to extract the metadata directly from redcap.

**Comment by fredbarthelemy** [comment]
FRED says:
_I agree that setting limits / responsabilities would make this much easier. I like the idea of the standard user not having to interact with redcap itself as this would make_ 
Last time we discussed and agreed on this point for survey conception, not for daily use of redcap.

_A) their life much easier,_ 

I don't think so. This will change their life from "I made a mistake and I can fix it immediatly' to 'I made a mistake and I have to make a ticket and harras the NIT to fix it'


_B) limit the number of people who need to know about redcap mechanics to a few people (experts / datamanager / fredish people)._
Also no. In a situation where a user would need to have access to the API to fix it, he should anyway know enough about the redcap mechanics to figure out if it is solvable. Letting the users with no knowledge of the functionning of a system is, in my opinion, the best way to make them 1)screw it up, 2) don't understand why they screwed, 3) complain about the system and loose confidence in it.

_Regarding the need for change of records later on there's two cases:
 - The current implementation has a feature (more likely a bug) that does not mark all records as complete that are complete. I hope this will be fixed by using a different mechanic after the RedCap server update.
 - A mistake during the recording. This should be rare and I think can be handled by approaching one redcap expert / datamanager to fix this single entry._

That's very optimistic
I have only two users and I already have: one missing session, one session added because they thought they will start a new one but ended just with the reward in it and no data and two commas in the comments that lead to errors in the CSV

_In this scenario the users would not need to need to use excel / libre office calc for loading the data, assuming they adopt the BIDS approach and do not need to extract the metadata directly from redcap._
Objectively, I start to think that a standardized excel sheet will fullfill all our requirement well better than redcap
